manager_dashboard/manager_dashboard/js/utils.js

```diff
@@ -0,0 +1,134 @@
+const getFormData = object => Object.keys(object).reduce((formData, key) => {
+    formData.append(key, object[key]);
+    return formData;
+}, new FormData());
+
+
+function geojsonToFeatureCollection(geojson){
+    if (geojson.type != "FeatureCollection"){
+        var collection = {
+            type: 'FeatureCollection',
+            features: [{"type":"feature", "geometry":geojson}]
+        };
+        return collection;
+    }
+    return geojson;
+}
+
+
+async function projectAoi(TMId){
+    let response = await fetch(`https://tasking-manager-tm4-production-api.hotosm.org/api/v2/projects/${TMId}/queries/aoi/?as_file=false`);
+    const answer = await response.json();
+    return answer
+}
+
+
+async function ohsome_request(url, data){
+    let response = await fetch(url, {
+        "method": 'POST',
+        "body": data,
+    })
+    const answer_ohsome = await response.json();
+    return answer_ohsome
+}
+
+
+async function countObjectsInFilter(bpolys, filter){
```

Comment:
Here or maybe in the function above we need to check if the ohsome api returns a valid response code. E.g I got the following:

```
he defined bpolys parameter contains some invalid geometry: found non-noded intersection between LINESTRING ( -73.1112670898438 -3.52986941895629, -73.0673217773438 -3.56002401255892 ) and LINESTRING ( -73.114013671875 -3.51890386902585, -73.0728149414063 -3.56002401255892 ) [ (-73.08489967365882, -3.5479623320379616, NaN) ]
```

So whenever the ohsome api count requests returns something different from 200 you should not be able to set up a project.

Reply:
-> should probably be addressed in utils

Reply:
same as other comment